### DISCUSSION TITLE
Possible to condense rows of Detail view in Explorer?

### DISCUSSION THREAD

**Discussion by VulcanTourist** [valinet/ExplorerPatcher#169] [Ideas]
Windows 11 has relaxed the line/row spacing in Explorer, compared to Windows 10.  I find it not only very distracting but inefficient, since less information is now displayed in the same viewport.  Can an option be added to adjust this?  Ideally this option would either also apply to the navigation pane (directory hierarchy) or there would be a separate option for adjusting the spacing there.

**Comment by Gaurav-Original-ClassicShellTester** [comment]
There is an official setting in File Explorer Options (Folder Options) -> View tab. Decrease space between items (compact view)

HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced
UseCompactView: 1 (Enabled)

**Reply by VulcanTourist** [reply]
I haven't visited those settings in years!  Is that a setting new to Windows 11?  I can't recall it from earlier.  If not, I wonder if it's possible that I had chosen it long before, and the Windows 11 installation reset it?

No matter; it functions as advertised!

**Reply by Gaurav-Original-ClassicShellTester** [reply]
Yes it's new to very recent releases of Windows 10 and to Windows 11 :)

**Reply by VulcanTourist** [reply]
The compaction must have been something I was accomplishing external to Windows 10 with another utility that no longer works....

**Reply by Gaurav-Original-ClassicShellTester** [reply]
Well there is another way to further compact the list and that is to force Explorer to use the SysListView32 control (used by Vista and earlier) instead of the new ItemsView that Windows 7 introduced. Folder Options X can do that programmatically without registry tweaks: https://github.com/T800G/FolderOptionsX but there is also a registry value per folder "template" (Docs, Videos, Music, Images etc) to make Explorer use the older and documented-for-use-by-apps ListView control instead of the private use ItemsView.

See the difference in compactness/spacing. Windows 7 tried to make certain things touch-oriented to some extent (but not as mad as Windows 8/10/11 where Microsoft shoves a one-size-fits-all touch-first UI down mouse & keyboard users' throat)

![image](https://user-images.githubusercontent.com/46422834/138437555-1a961428-b3f7-4bed-ae15-1a7fde5e7244.png)

**Reply by VulcanTourist** [reply]
I already had RegEdit "Favorites" shortcuts to those templates from earlier tinkering with them years ago, but I don't see any existing obvious route to leveraging this ListView method.  Also, the FolderOptionsX Github page makes no mention of this as a feature.  I am left without an obvious way to make an informed decision without running off down a couple rabbit holes.  I have too many half-explored rabbit holes as it is.

**Reply by Gaurav-Original-ClassicShellTester** [reply]
Checking any of the first 3 options in Folder Options X enables the older ListView (except the last option which is for the nav. pane/tree view control). In fact the Registry tweak is a bit of a nightmare since you have to change it for all the templates and still certain folders are not changed due to Explorer's "Remember each folder's view distinctly if it is customized" behavior which btw is very nice.

**Reply by VulcanTourist** [reply]
> Checking any of the first 3 options....

Does that feller need to be a better documentarian?  I couldn't have inferred this from reading the Github main page, as you did from digging into the source code.

> ... due to Explorer's "Remember each folder's view distinctly if it is customized" behavior....

I've relied on that frequently, too, so perhaps I don't want to jeopardize its value.  It would seem that Folder Options X is the means for more compaction if I conclude that I need it consistently everywhere.  OTOH, using the Registry would allow me to specify extra compaction for certain content templates exclusively; I haven't thought of examples yet where I would need that, but what property would I need to add to enable ListView selectively like that?

**Reply by Gaurav-Original-ClassicShellTester** [reply]
Well this is completely untested and I don't remember where I found it on the web. Probably winhelponline.com since Ramesh Srinivasan was very early to find it, I remember that much.

Please do a full registry backup using ERUNT, or System Restore or exporting the below-modified reg keys before merging it. Untested regedits can have weird side-effects so prior backup is must.

Windows Registry Editor Version 5.00

;Brings Back Auto Arrange Menu Option in Windows 7
;Reduces spacing of items in right pane of Explorer
;Turns on column headers for all views

[-HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags]

[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Advanced]
"FullRowSelect"=dword:00000000

[HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags\AllFolders\Shell]
"KnownFolderDerivedFolderType"="{57807898-8C4F-4462-BB63-71042380B109}"
"SniffedFolderType"="Generic"
"FolderType"="NotSpecified"

;Generic - Folder Template

[HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags\AllFolders\Shell\{5C4F28B5-F869-4E84-8E60-F11DB97C5CC7}]
"Rev"=dword:00000000
"FFlags"=dword:43000000
"Vid"="{0057D0E0-3573-11CF-AE69-08002B2E1262}"
"Mode"=dword:00000001
"LogicalViewMode"=dword:00000001
"IconSize"=dword:00000010
"GroupView"=dword:00000000
"GroupByKey:FMTID"="{00000000-0000-0000-0000-000000000000}"
"GroupByKey:PID"=dword:00000000
"GroupByDirection"=dword:00000001

;Documents - Folder Template

[HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags\AllFolders\Shell\{7d49d726-3c21-4f05-99aa-fdc2c9474656}]
"Rev"=dword:00000000
"Vid"="{0057D0E0-3573-11CF-AE69-08002B2E1262}"
"Mode"=dword:00000001
"LogicalViewMode"=dword:00000001
"IconSize"=dword:00000010
"GroupView"=dword:00000000
"GroupByKey:FMTID"="{00000000-0000-0000-0000-000000000000}"
"GroupByKey:PID"=dword:00000000
"GroupByDirection"=dword:00000001
"FFlags"=dword:43000000

;Music - Folder Template

[HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags\AllFolders\Shell\{94d6ddcc-4a68-4175-a374-bd584a510b78}]
"Rev"=dword:00000000
"Vid"="{0057D0E0-3573-11CF-AE69-08002B2E1262}"
"Mode"=dword:00000001
"LogicalViewMode"=dword:00000001
"IconSize"=dword:00000010
"GroupView"=dword:00000000
"GroupByKey:FMTID"="{00000000-0000-0000-0000-000000000000}"
"GroupByKey:PID"=dword:00000000
"GroupByDirection"=dword:00000001
"FFlags"=dword:43000000


;Pictures - Folder Template

[HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags\AllFolders\Shell\{b3690e58-e961-423b-b687-386ebfd83239}]
"Rev"=dword:00000000
"Vid"="{0057D0E0-3573-11CF-AE69-08002B2E1262}"
"Mode"=dword:00000001
"LogicalViewMode"=dword:00000001
"IconSize"=dword:00000010
"GroupView"=dword:00000000
"GroupByKey:FMTID"="{00000000-0000-0000-0000-000000000000}"
"GroupByKey:PID"=dword:00000000
"GroupByDirection"=dword:00000001
"FFlags"=dword:43000000

;Videos - Folder Template

[HKEY_CURRENT_USER\Software\Classes\Local Settings\Software\Microsoft\Windows\Shell\Bags\AllFolders\Shell\{5fa96407-7e77-483c-ac93-691d05850de8}]
"Rev"=dword:00000000
"Vid"="{0057D0E0-3573-11CF-AE69-08002B2E1262}"
"Mode"=dword:00000001
"LogicalViewMode"=dword:00000001
"IconSize"=dword:00000010
"GroupView"=dword:00000000
"GroupByKey:FMTID"="{00000000-0000-0000-0000-000000000000}"
"GroupByKey:PID"=dword:00000000
"GroupByDirection"=dword:00000001
"FFlags"=dword:43000000
